idanalysis/eqparams.py

```diff
@@ -0,0 +1,1194 @@
+        e0 = mathphys.constants.vacuum_permitticity
+        u0 = (
+            1e-3
+            * (
+                (e * gamma**4)
+                * cumtrapz(dx=dx, y=(b / brho) ** 2)
+                / (6 * _np.pi * e0)
+            )[-1]
+        )
+        self._u0 = u0
+        return u0
+
+    def calc_power(self, current=0.1):
+        """Calculate radiated power in ID.
+
+        Args:
+            current (float, optional): Current in [A]. Defaults to 0.1.
+
+        Returns:
+            float: Power in [kW]
+        """
+        p = current * self.calc_u0()
+        return p
+
+    def calc_i1(self):
+        """Calculate first radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I1x
+            float: delta I1y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        etax, etay = self.calc_dispersion()
+        i1x = cumtrapz(dx=dx, y=etax * by[1:-1] / self.brho)[-1]
+        i1y = cumtrapz(dx=dx, y=etay * bx[1:-1] / self.brho)[-1]
+        self._i1x = i1x
+        self._i1y = i1y
+        return i1x, i1y
+
+    def calc_i2(self):
+        """Calculate second radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I2x
+            float: delta I2y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        i2x = cumtrapz(dx=dx, y=(by / self.brho) ** 2)[-1]
+        i2y = cumtrapz(dx=dx, y=(bx / self.brho) ** 2)[-1]
+        self._i2x = i2x
+        self._i2y = i2y
+        return i2x, i2y
+
+    def calc_i3(self):
+        """Calculate third radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I3x
+            float: delta I3y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        i3x = cumtrapz(dx=dx, y=_np.abs((by / self.brho) ** 3))[-1]
+        i3y = cumtrapz(dx=dx, y=_np.abs((bx / self.brho) ** 3))[-1]
+        self._i3x = i3x
+        self._i3y = i3y
+        return i3x, i3y
+
+    def calc_i4(self):
+        """Calculate fourth radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I4x
+            float: delta I4y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        x = 1e-3 * self.field_profile[:, 0]
+        dx = _np.diff(x)[0]
+        dby = _np.diff(by)
+        dbx = _np.diff(bx)
+        px = cumtrapz(dx=dx, y=by) / self.brho
+        py = cumtrapz(dx=dx, y=-bx) / self.brho
+        kx = -(dby / dx) * px / self.brho
+        ky = -(dbx / dx) * py / self.brho
+        etax, etay = self.calc_dispersion()
+        i4x = cumtrapz(
+            dx=dx,
+            y=(
```

Comment:
do you think that this is more readable? I prefer calculating `y` in a separate line...